From 4c6c00cfa9e95ca2e10bc20de261f9a92afad584 Mon Sep 17 00:00:00 2001
From: Thomas Bourgoin <thomas.bourgoin@foss.st.com>
Date: Thu, 11 Jan 2024 19:00:28 +0100
Subject: [PATCH 0770/1141] crypto: stm32/cryp - fix debugfs reg dump

Fill dev property of debugfs_regset32 struct to call the function
pm_runtime_get_sync before reading the registers of cryp.
Fixing SERC exception of BZ : 168813

Change-Id: I81ed9c384e1ef8bf6b0bf9ba2c1c87d16f568fcf
Signed-off-by: Thomas Bourgoin <thomas.bourgoin@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/350820
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
Reviewed-by: Yann GAUTIER <yann.gautier@foss.st.com>
---
 drivers/crypto/stm32/stm32-cryp.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/crypto/stm32/stm32-cryp.c b/drivers/crypto/stm32/stm32-cryp.c
index 283ec1469b6b..74fb762af804 100644
--- a/drivers/crypto/stm32/stm32-cryp.c
+++ b/drivers/crypto/stm32/stm32-cryp.c
@@ -2358,6 +2358,7 @@ static void cryp_debugfs(struct stm32_cryp *cryp)
 	if (!regset)
 		return;
 
+	regset->dev = cryp->dev;
 	regset->regs = stm32_cryp_regs;
 	regset->nregs = ARRAY_SIZE(stm32_cryp_regs);
 	regset->base = cryp->regs;
-- 
2.39.2

